### ISSUE TITLE
Parameterize footnote links

### ISSUE THREAD

**Issue by ns-cweber** [russross/blackfriday#269]
I've written a static blog generator which generates post snippets on index pages. Sometimes these snippets contain footnote markers which link to `#fn:<label>` which the browser interprets as `index.html#fn:<label>`. I would like to be able to modify the target file for these links so they go to `some-post-filename.html#fn:<label>`; however, I don't believe there is support for that at the moment. It would be nice to be able to parameterize the HTMLRenderer to specify a filename prefix for the anchors so the footnote links go to the correct page, even if that isn't the current page.

**Comment by ns-cweber** [comment]
It looks like this would be a two line change to `html.go` (if this is acceptable, I can add test cases, improve the parameter name, and submit a PR):

```diff
diff --git a/html.go b/html.go
index 74e67ee..41e46ca 100644
--- a/html.go
+++ b/html.go
@@ -61,6 +61,9 @@ type HtmlRendererParameters struct {
 	AbsolutePrefix string
 	// Add this text to each footnote anchor, to ensure uniqueness.
 	FootnoteAnchorPrefix string
+	// Add this text before the '#' on the footnote anchor, useful if the
+	// footnote body is in a different file
+	FootnoteAnchorURLPrefix string
 	// Show this text inside the <a> tag for a footnote return link, if the
 	// HTML_FOOTNOTE_RETURN_LINKS flag is enabled. If blank, the string
 	// <sup>[return]</sup> is used.
@@ -619,7 +622,10 @@ func (options *Html) FootnoteRef(out *bytes.Buffer, ref []byte, id int) {
 	out.WriteString(`fnref:`)
 	out.WriteString(options.parameters.FootnoteAnchorPrefix)
 	out.Write(slug)
-	out.WriteString(`"><a rel="footnote" href="#`)
+	out.WriteString(fmt.Sprintf(
+		`"><a rel="footnote" href="%s#`,
+		options.parameters.FootnoteAnchorURLPrefix,
+	))
 	out.WriteString(`fn:`)
 	out.WriteString(options.parameters.FootnoteAnchorPrefix)
 	out.Write(slug)
```

**Comment by shurcooL** [comment]
> Sometimes these snippets contain footnote markers which link to `#fn:<label>` which the browser interprets as `index.html#fn:<label>`.

Just to clarify, but the brower should interpret that as `currentpage.html#fn:<label>`, where `currentpage.html` is the last path element of the current URL. Is that right?

**Comment by ns-cweber** [comment]
@shurcooL Correct. In my use case, the index page which lists all of the snippets is the current page, but I would like them to link to the page which contains the full post body (including the footnote definitions).

**Comment by ns-cweber** [comment]
@shurcooL Does this look like something that could be added? I'm willing to submit a PR; I'd just rather not do the work if it will get rejected.

**Comment by ns-cweber** [comment]
@shurcooL I largely agree with your concerns, but I imagine "post snippets" is a sufficiently common use case that it should be supported somehow. Leaving broken links in the snippet body is pretty bad, and having to munge the output HTML really detracts from this library's appeal. I'm open to other suggestions in case this one is too specific; it would just be very nice if this could be addressed in this library.

**Comment by rtfb** [comment]
First, consider trying out v2. Things like that should already be possible there on the user side. And if they're not, I'd love to add what's missing.

If you can't switch to v2 for some reason, then go ahead with the patch, it doesn't seem to hurt.
